### PR TITLE
Management UI for compositions

### PR DESCRIPTION
**Pull request by warpech**
Implements https://github.com/Starcounter/Blending/issues/316

The changes are explained in the commit messages, so here I will just paste the screenshots.

This PR will be the basis for #301

2018-10-30 08_18_47-blendingeditor
2018-10-30 08_18_58-blendingeditor

### CODE REVIEW

File: src/BlendingEditor/BlendingEditor.csproj [hunk 1]
```diff
@@ -58,6 +58,9 @@
       <HintPath>$(SolutionDir)\packages\Starcounter.Linq.2.2.0\lib\net461\Starcounter.Linq.dll</HintPath>
     </Reference>
     <Reference Include="Starcounter.QueryProcessor, Version=1.0.0.0, Culture=neutral, PublicKeyToken=d2df1e81d0ca3abf, processorArchitecture=MSIL" />
+    <Reference Include="Starcounter.Uniform, Version=2.1.1.0, Culture=neutral, processorArchitecture=AMD64">
+      <HintPath>..\..\packages\Starcounter.Uniform.2.1.1\lib\net461\Starcounter.Uniform.dll</HintPath>
```
Comment (PatrykSzwer) [inline src/BlendingEditor/BlendingEditor.csproj]: Wrong package path, please add solutionDir.

File: src/BlendingEditor/wwwroot/BlendingEditor/views/CompositionsPage.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        <div class="uni-section-secondary uni-title">
+            <slot name="blendingeditor/compositions-header"></slot>
+        </div>
+        <div class="uni-section-primary blendingeditor-section-primary-grid">
+            <div class="uni-card">
+                <slot name="blendingeditor/compositions-introduction"></slot>
+                <details>
+                    <summary>Help</summary>
+                    <div>
+                        <p>For more information about view composing, <a href="https://docs.starcounter.io/guides/web-apps/view-composing" target="_blank">read the docs</a>.</p>
```
Comment (PatrykSzwer) [inline src/BlendingEditor/wwwroot/BlendingEditor/views/CompositionsPage.html]: Clicking on this link returns "Page not found". Please fix the link.

File: src/BlendingEditor/wwwroot/BlendingEditor/views/CompositionPage.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        <td class="blendingeditor-wrappable-text">{{attachment.AttachedView.Uri}}</td>
+                    </tr>
+                </template>
+            </table>
+
+            <juicy-ace-editor slot="blendingeditor/composition-editor"
+                              mode="ace/mode/html"
+                              wrapmode="true" value="{{model.Composition$::change}}"></juicy-ace-editor>
+
+            <button slot="blendingeditor/composition-save" onclick="++this.value;" value="{{model.SaveTrigger$::click}}">Save</button>
```
Comment (PatrykSzwer) [inline src/BlendingEditor/wwwroot/BlendingEditor/views/CompositionPage.html]: Save and cancel buttons are always active. Please add transaction dirty check to disable those buttons when there are no changes.
Also please add Back button to go back to compositions list.

**Comment by warpech** [comment]
All addressed now, but the Cancel button needs to be clicked twice to become disabled again.

This problem is tracked in a separate issue https://github.com/Starcounter/Blending/issues/319

@PatrykSzwer please re-review here.

File: src/BlendingEditor/Website/ViewModels/CompositionPage.json.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@ static CompositionPage()
             DefaultTemplate.Composition.Bind = "Html";
             DefaultTemplate.SaveTrigger.Bind = null;
             DefaultTemplate.CancelTrigger.Bind = null;
+            DefaultTemplate.Trn.Bind = null;
```
Comment (PatrykSzwer) [inline src/BlendingEditor/Website/ViewModels/CompositionPage.json.cs]: I believe that this is not needed (probably for almost all of those here), but please check and make sure.

Comment (warpech) [reply src/BlendingEditor/Website/ViewModels/CompositionPage.json.cs]: It is needed because I used `IExplicitBound`

**Pull request merged**